### PR TITLE
Correction to the mode decomposition

### PR DESCRIPTION
**Pull request by Paaaaarth**
We have opened this pull request to modify the mode decomposition in order to make it more versatile. We have removed the dependence on field_interp which is a function of transverse profile from data class and not of the base transverse profile class.

### CODE REVIEW

File: lasy/utils/mode_decomposition.py [hunk 1]
```diff
@@ -9,7 +9,15 @@
...
 
-def hermite_gauss_decomposition(laserProfile, wavelength, m_max=12, n_max=12, res=1e-6):
+def hermite_gauss_decomposition(
+    laserProfile,
+    wavelength,
+    m_max=12,
+    n_max=12,
+    res=1e-6,
+    lo=[-2e-4, -2e-4],
+    hi=[2e-4, 2e-4],
```
Comment (MaxThevenet) [inline lasy/utils/mode_decomposition.py]: Could you make these 3 mandatory parameters?

**Review by MaxThevenet — CHANGES_REQUESTED** [review]
Thanks for this PR! See comment below.

File: lasy/utils/mode_decomposition.py (hunk 1 above)
Comment (Paaaaarth) [reply lasy/utils/mode_decomposition.py]: Thanks you for the feedback. The changes suggested have been made.

Comment (Paaaaarth) [reply lasy/utils/mode_decomposition.py]: Thank you for the feedback:)
The requested changes have been made. Additionally n_max and m_max also needed to be made mandatory in order to keep the code as consistent as possible.

File: lasy/utils/mode_decomposition.py [hunk 2]
```diff
@@ -9,7 +9,15 @@
 from lasy.utils.exp_data_utils import find_d4sigma
 
 
-def hermite_gauss_decomposition(laserProfile, wavelength, m_max=12, n_max=12, res=1e-6):
+def hermite_gauss_decomposition(
+    laserProfile,
+    wavelength,
+    m_max,
+    n_max,
```
Comment (MaxThevenet) [inline lasy/utils/mode_decomposition.py]: I think it's good to keep the default 12 for these two, actually, people may wonder what makes sense. That indeed requires one change in the order of parameters in the doc.

**Review by MaxThevenet — CHANGES_REQUESTED** [review]

File: lasy/utils/mode_decomposition.py (hunk 2 above)
Comment (Paaaaarth) [reply lasy/utils/mode_decomposition.py]: Requested changes have been made:)

**Review by MaxThevenet — APPROVED** [review]
Thanks!

**Pull request merged**